src/zeit/content/volume/browser/toc.py

```diff
@@ -0,0 +1,413 @@
+# -*- coding: utf-8 -*-
+import StringIO
+import csv
+import urlparse
+import re
+import sys
+import posixpath
+from ordereddict import OrderedDict
+
+import zeit.cms.browser.view
+from zeit.cms.i18n import MessageFactory as _
+import zeit.cms.content.sources
+import zeit.cms.interfaces
+from zeit.cms.repository.interfaces import IFolder
+import zeit.connector.connector
+from zeit.content.article.interfaces import IArticle
+from zeit.content.volume.interfaces import ITocConnector
+
+import zope.app.appsetup.product
+import zope.component
+import zope.component.registry
+import zope.interface
+import zope.site.site
+
+import toc_config
+
+class Toc(zeit.cms.browser.view.Base):
+    """
+    View for creating a Table of Content as a csv file.
+    The dav url to the articles generally looks like
+    ARCHIVE_ROOT/PRODUCT_ID/RESSORT_NAME/ARTICLE.xml
+    E.g.
+    cms-backend.zeit.de/cms/archiv-wf/archiv/ZEI/2016/23/entdecken/03-Normalo
+    """
+    # Christ und Welt product ID is a special case,
+    # because in the products.xml the deprecated product ID 'CW' is present.
+    PRODUCT_ID_DIR_NAME_EXCEPTIONS = {'CW': 'ZECW'}
+    CSV_DELIMITER = '\t'
+
+    def __init__(self, *args, **kwargs):
+        super(Toc, self).__init__(*args, **kwargs)
+        config = zope.app.appsetup.product\
+            .getProductConfiguration('zeit.content.volume')
+        self.dav_archive_url = config.get('dav-archive-url')
+        self.dav_archive_url_parsed = self._parse_config()
+        self.excluder = Excluder()
+        self._register_archive_connector()
+        self.connector = zope.component.getUtility(ITocConnector)
+
+    def _parse_config(self):
+        """
+        Parse DAV-Archive URL.
+        :return: urlparse.ParseResult
+        """
+        return urlparse.urlparse(self.dav_archive_url)
+
+    def _register_archive_connector(self):
+        """Register the ITocConnecor utility if necessary"""
+        if self.dav_archive_url == 'test':
```

Comment:
Hmm, es wär glaub ich effizienter, wenn man per ZCML (dann muss man halt wohl wirklich getrennte Dateien machen, mock.zcml und real.zcml, und die in ftesting.zcml bzw. site.zcml laden) den zweiten Connector _einmal_ instanziert, und dann weiter unten jeweils nochmal als IConnector registriert. Dann hat man nur 1 Objekt, statt immer pro Request ein neues.